### DISCUSSION TITLE
Support for C++

### DISCUSSION THREAD

**Discussion by IvanovichPushkin** [starship/starship#5635] [Ideas]
Will there be an update for Starship that will allow support for C++?

**Comment by jankatins** [comment]
What do you mean with "support for C++". 

If you ask if you can write additional modules for starship in c++, I would doubt that: starship is written in rust and adding the complexity to integrate modules written in another language seems overkill to me.

If you want to show specific information for your c++ development environment, then this should be possible, e.g. a lot of other dev env thingies are shown: compiler and interpreter versions, cmake,... What are you missing?

**Reply by IvanovichPushkin** [reply]
I think what you said in the latter was what I meant. I've noticed that when using that when I was accessing a directory with c, python, or rust files. The prompt would show indication that there these are present. On the other hand, that was not the case with c++.

**Reply by mickimnet** [reply]
If your _C++_ compiler / setup uses different commands, you can add / change them in the [C module section](https://starship.rs/config/#c) of the configuration.

**Reply by IvanovichPushkin** [reply]
What about unique symbols for a c++ setup? Is it possible for that to happen for starship?

**Reply by mickimnet** [reply]
Yes, someone can write a module for that. Or you could create a custom module configuration for it, yourself.

**Reply by IvanovichPushkin** [reply]
thanks. also btw I made a module for it however the following message appears:
[WARN] - (starship::config): Error in 'Custom' at 'version_format': Unknown key

do you know what's the problem with my code. Also here is my code:

[custom.cpp]
detect_extensions = ['cpp']
symbol = "󰙲 "
format = 'via [$symbol($version(-$name) )]($style)'
version_format = 'v${raw}'
commands = [ [ 'g++', '--version' ] ]
style = 'bold fg:navy'

**Reply by knockout101** [reply]
I just wanted to add to this. To get this table to trigger inside the format string, you have to use "$custom\".

format = """
[](#27abff)\
$directory\
[](fg:#27abff bg:#31c4ff)\
$git_branch\
$git_status\
[](fg:#31c4ff bg:#41dfff)\
$c\
$custom\   <<<
$elixir\
$elm\
$golang\
$gradle\
$haskell\
$java\
$julia\
$nodejs\
$nim\
$rust\
$scala\
$python\
[](fg:#41dfff bg:#7ef8ff)\
$docker_context\
[](fg:#7ef8ff bg:#bbfdff)\
$time\
[ ](fg:#bbfdff)\
"""

**Reply by IvanovichPushkin** [reply]
I don't get this, what's the "<<<" for?

**Reply by IvanovichPushkin** [reply]
also the commands don't seem  to work

**Comment by harisonkhlil** [comment]
> thanks. also btw I made a module for it however the following message appears: [WARN] - (starship::config): Error in 'Custom' at 'version_format': Unknown key
> 
> do you know what's the problem with my code. Also here is my code:
> 
> [custom.cpp] detect_extensions = ['cpp'] symbol = "󰙲 " format = 'via [$symbol($version(-$name) )]($style)' version_format = 'v${raw}' commands = [ [ 'g++', '--version' ] ] style = 'bold fg:navy'


mine is ok:
```toml
[custom.cpp]
detect_extensions = ['cpp', 'cc', 'hpp']
symbol = "󰙲 "
format = "via [$symbol($output)]($style)"
command = "clang++ --version | awk '/clang version/{print $4}'"
style = "bold 149"
disabled = false
ignore_timeout = false
```

**Reply by IvanovichPushkin** [reply]
ok thanks although i could not get it to output the version in the prompt for g++

**Reply by niccentury** [reply]
I did this and it worked to get the version for g++ 

```
format = """
...
$custom\
...
"""

[custom.cpp]
detect_extensions = ['cpp', 'hpp']
symbol = " "
style = "bg:#86BBD8"
format = '[ $symbol ($output) ]($style)'
command = "g++ --version | awk 'NR==1 {print $3}'"
```

**Reply by IvanovichPushkin** [reply]
ok it now works, thanks

**Comment by ahmadsimran** [comment]
I have written this and it worked 

[custom.cpp]
detect_extensions = ['c', 'h', 'cpp', 'cc', 'hpp']
symbol = "C="
style = "bg:#0373fc"
format = '[$symbol ($output) ]($style)'
command = "g++ --version | head -n1 | cut -d' ' -f4 | cut -d'-' -f1 | cut -d'.' -f1,2"

Also add it in the layout at top of toml file
